query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs

```diff
@@ -44,7 +43,7 @@ pub struct ChainingHashTable {
     ctx: Arc<QueryContext>,
 
     /// A shared big hash table stores all the rows from build side
-    hash_table: RwLock<Vec<Vec<RowPtr>>>,
+    hash_table: RwLock<HashMap<KeysRef, Vec<RowPtr>>>,
```

Comment:
KeysRef maybe memory leak?

Reply:
The usage of `KeysRef` in hash join without unsafe. Could you please explain why you think `KeysRef` may cause memory leak? Thanks~

Reply:
We referenced the data by ptr, but the data has been destroyed with the iteration of the data block.